posthog/api/test/test_csp.py

```diff
@@ -0,0 +1,286 @@
+import json
+
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+from posthog.api.csp import (
+    process_csp_report,
+    parse_report_uri,
+    parse_report_to,
+    parse_properties,
+)
+
+
+class TestCSPModule(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_parse_report_uri(self):
+        """Test parsing CSP report in report-uri format"""
+        data = {
+            "csp-report": {
+                "document-uri": "https://example.com/foo/bar",
+                "referrer": "https://www.google.com/",
+                "violated-directive": "default-src self",
+                "effective-directive": "img-src",
+                "original-policy": "default-src 'self'; img-src 'self' https://img.example.com",
+                "disposition": "enforce",
+                "blocked-uri": "https://evil.com/malicious-image.png",
+                "line-number": 10,
+                "source-file": "https://example.com/foo/bar.html",
+                "status-code": 0,
+                "script-sample": "alert('hello')",
+            }
+        }
+
+        properties = parse_report_uri(data)
+
+        self.assertEqual(properties["$current_url"], "https://example.com/foo/bar")
+        self.assertEqual(properties["violated_directive"], "default-src self")
+        self.assertEqual(properties["blocked_url"], "https://evil.com/malicious-image.png")
+        self.assertEqual(properties["script_sample"], "alert('hello')")
```

Comment:
do we need to make sure we encode here? or really is it already encoded when we get it

someone could take advantage of this field?

Reply:
We could encode, hash, or redact it. I think that redacting or dropping it might be better.

Reply:
Used django escape while we decide if we wan't to redact it.